### ISSUE TITLE
Run Matlab tests in test step of build script

### ISSUE THREAD

**Issue by hsautessella** [pace-neutrons/Herbert#33]
The build script `tools/build_config/build.sh` created in #30 does not currently run Matlab tests. It should be edited to do so.

In order for the tests to work with Jenkins Matlab will need to exit on a non-zero error code if any tests fail. This blog post may be useful https://blogs.mathworks.com/developer/2015/01/20/the-other-kind-of-continuous-integration/

**Comment by abuts** [comment]
I do not think it should exit on non-0 code (unless you hope to achieve to run Matlab in Parallel which I would not advice as initial step. The blog refers to different unit test suite from one we use. I have not read the full blog yet, but our test suite generates reports and write results for all tests package. 


Try runtests "test_folder" 
runtests test_file
or validate_herbert 
first from Matlab to try the existing testing procedure. 
then you can include any level of this type of testing in the script

**Comment by hsautessella** [comment]
The reason I said that it needed to exit on a non-zero code is that: how does Jenkins know that a test has failed?

The tests will probably be run from within the `build.sh` script, using a command like `matlab -nodisplay -nosplash -r "validate_herbert; exit"`. But as far as Jenkins is concerned, the Matlab process it spawns will start and exit normally and the build will pass. Exiting with a non-zero code tells Jenkins that something has gone wrong and that the build should fail.

Or am I missing something?

**Comment by abuts** [comment]
you need to modify validate_horace/herbert script to accept -run_as_script option and if this option is provided 
use `quit` (see help `quit`) with code depending on the results of the matlab unit tests. 
then you do not need to add exit to the end of the execution script as described in the blog
